src/main/webapp/creatives/maps/google_maps_client.js

```diff
@@ -0,0 +1,260 @@
+         * @private {?Element}
+         */   
+        this.timeDisplayElement_ = null;
+    }
+
+    /**
+     * Assigns the proper HTML element representing the 
+     * travel method selector to a variable that can be used inside
+     * of the Simid Map class.
+     * @param {?Element} element An HTML element.
+     */
+    setTravelMethodElement(element) {
+        this.travelMethodElement_ = element;
+    }
+
+    /**
+     * Assigns the proper HTML element representing the 
+     * time display div to a variable that can be used inside
+     * of the Simid Map class.
+     * @param {?Element} element An HTML element.
+     */
+    setTimeDisplayElement(element) {
+        this.timeDisplayElement_ = element;
+    }
+
+    /**
+     * Returns the map object within the Simid Map class. 
+     */
+    getMap() {
+        return this.map_;
+    }
+
+    /**
+     * Loads a map object that currently defaults to a hardcoded location.
+     * @param {!Element} mapElement The div within the main document where the map is to be displayed.
+     */
+    displayMap(mapElement) {
+        this.map_ = new google.maps.Map(mapElement, {
+            zoom: DEFAULT_ZOOM,
+            center: this.currentLocation_
+        });
+        this.addMapListener_();
+        this.findNearby_();
+    }
+
+    /**
+     * Searches for the closest corresponding locations based off of the given search parameter,
+     * and places pins on the map that represent the closest locations.
+     * @private
+     */
+    findNearby_() {
+        const request = {
+            location: this.currentLocation_,
+            name: this.searchQuery_,
+            openNow: true,
+            rankBy: google.maps.places.RankBy.DISTANCE
+        };
+        const placeService = new google.maps.places.PlacesService(this.map_);
+        placeService.nearbySearch(request, this.displayResults_.bind(this));
+    }
+
+    /**
+     * Displays the closest advertisement's locations to a user's current location.
+     * @param {!Array<!Object>} results An array of Place Results from the search query.
+     * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+     *  by the PlacesService on the completion of its searches.
+     * @private 
+     */
+    displayResults_(results, status) {
+        if (status == google.maps.places.PlacesServiceStatus.OK) {
+            //Active location is set to the closest location to start.
+            this.activeLocation_ = results[0].geometry.location;
+            for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
+                this.placeMapMarker_(results[i]);
+            }
+            this.displayDirections_();
+        } else {
+            const statusErrorMessage = {
+                message: "ERROR: Failed to complete search: " + status,
+            };
+            this.simidProtocol.sendMessage(CreativeMessage.LOG, statusErrorMessage);
+            //this.playAd_();
```

Comment:
You have to decide how to do this.

A few ways come to mind.
1. Have event handlers.
2. Pass in a callback function.

Reply:
Handled this in our meeting